src/transform.js

```diff
@@ -278,7 +278,8 @@ export default class i18nTransform extends Transform {
 
         if (this.options.failOnUpdate) {
           const addCount = uniqueCount[namespace] - mergeCount
-          if (addCount + restoreCount + oldCount !== 0) {
+          const unreferencedCount = this.options.keepRemoved ? oldCount : 0;
+          if (addCount + restoreCount + oldCount - unreferencedCount !== 0) {
```

Comment:
This doesn't account for when `keepRemoved` is not a boolean. We use a regex to match 'common' namespace translations, so that parsing any package in our repo doesn't remove old `common:` translations.

```ts
  keepRemoved: [/^common:/],
```

Reply:
@RobHannay Feel free to open a new PR with proper testing.